Sources/TuistCore/MetadataProviders/XCFrameworkMetadataProvider.swift

```diff
@@ -75,12 +77,23 @@ public final class XCFrameworkMetadataProvider: PrecompiledMetadataProvider, XCF
             primaryBinaryPath: primaryBinaryPath,
             linking: linking,
             mergeable: infoPlist.libraries.allSatisfy(\.mergeable),
-            status: status
+            status: status,
+            macroPath: try macroPath(xcframeworkPath: path)
         )
     }
 
+    /**
+     An XCFramework that contains static frameworks that represent macros, those are frameworks with a Macros directory in them.
+     We assume that the Swift Macros, which are command line executables, are fat binaries for both architectures supported by macOS:
+     x86_64 and arm64.
+     */
+    public func macroPath(xcframeworkPath: AbsolutePath) throws -> AbsolutePath? {
+        guard let frameworkPath = fileHandler.glob(xcframeworkPath, glob: "*/*.framework").sorted().first else { return nil }
```

Comment:
Is there always just one directory with a framework in the macro xcframework? Non-macro iOS targets have one for a device and one for simulator, but it might be that macros have really just one (device for macOS).

Reply:
> Is there always just one directory with a framework in the macro xcframework?

If the XCFramework represents a framework, yes. If it represents a library, which can also be the case, then there's no framework. This will reveal that there's no Swift Macro, which is accurate.
If there are multiple frameworks (e.g., device and simulator), both have the same copy of the macro, which is unnecessary duplication within the XCFramework, but I don't think it's worth adding additional logic to make sure there's only one because that'll make our logic a bit hard to reason about.